Src/Couchbase.Linq/Proxies/DocumentProxyGenerationHook.cs

```diff
@@ -51,7 +51,7 @@ public bool ShouldInterceptMethod(Type type, MethodInfo methodInfo)
             // Only proxy getters and setters for properties on the document
             // We must proxy getters or serializing the document back to Couchbase has problems
 
-            return methodInfo.IsSpecialName;
+            return methodInfo.IsSpecialName && methodInfo.Name.StartsWith("set_") || methodInfo.Name.StartsWith("get_");
```

Comment:
Jeff,

I think there's an operator precedence issue here if you don't wrap the || clause in parentheses.